kartothek/io_components/metapartition.py

```diff
@@ -183,6 +183,29 @@ class MetaPartition(Iterable):
     about the parent dataset
     """
 
+    _METADATA_COLUMNS = (
+        "partition_label",
+        "row_group_id",
+        "row_group_byte_size",
+        "number_rows_total",
+        "number_row_groups",
+        "serialized_size",
+        "number_rows_per_row_group",
+    )
+
+    _METADATA_DTYPES = (
+        np.dtype("O"),
+        np.dtype(int),
+        np.dtype(int),
+        np.dtype(int),
+        np.dtype(int),
+        np.dtype(int),
+        np.dtype(int),
+    )
+
+    MetadataSchema = namedtuple("MetadataSchema", _METADATA_COLUMNS)  # type: ignore
+    _METADATA_SCHEMA = MetadataSchema(**dict(zip(_METADATA_COLUMNS, _METADATA_DTYPES)))  # type: ignore
```

Comment:
Idea here was to store the "schema" of the returned statistics as something immutable, preferably something like a `frozendict` mapping column names to the dtypes.
Not sure, if this constructions is the best way to solve that.

Reply:
I wouldn't go for frozen(anything) since these containers are not as well supported as the default containers. For instance, I've seen occasionally serialization errors when using frozensets. What would be the value of defining this as frozen? I consider the risk of a user mutating this structure as *very* low and I would prefer the reduced complexity of not using special containers